jcl/src/java.base/share/classes/java/lang/invoke/ReceiverBoundHandle.java

```diff
@@ -90,11 +90,11 @@ boolean canRevealDirect() {
 	 */
 	/*[ENDIF]*/
 	MethodHandle permuteArguments(MethodType permuteType, int... permute) throws NullPointerException, IllegalArgumentException {
-		MethodHandle result = new PermuteHandle(permuteType, this, permute);
-		if (true) {
-			result = combinableVersion.permuteArguments(permuteType, permute);
+		if (isUnnecesssaryPermute(permuteType, permute)) {
```

Comment:
This is duplicate work as the same check will occur in the `combinableVersion.permuteArguments()` call